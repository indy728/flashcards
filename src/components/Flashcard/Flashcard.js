import React, { Component } from 'react'
import styled from 'styled-components'

const Wrapper = styled.div`
    perspective: 150rem;
    -moz-perspective: 150rem;
    position: relative;
    height: 52rem;

    :hover .flashcard-front{
        transform: rotateY(180deg);
    }
    :hover .flashcard-back {
        transform: rotateY(0deg);
    }
`

const FlashcardSide = styled.div`
    height: 52rem;
    transition: all .8s ease;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    -webkit-backface-visibility: hidden;
    backface-visibility: hidden;
    border-radius: 3px;
    overflow: hidden;
    box-shadow: 0 1.5rem 4rem rgba(0,0,0,.15);

`

const FlashcardFront = styled(FlashcardSide)`
    background-color: blueviolet;

`

const FlashcardBack = styled(FlashcardSide)`
    transform: rotateY(180deg);
    background-color: orangered;

`

class Flashcard extends Component {
    state = {
        reveal: false
    }

    render() {
        return (
            <Wrapper className='flashcard'>
                <FlashcardFront className='flashcard-front' />
                <FlashcardBack className='flashcard-back' />
            </Wrapper>
        )
    }
}

export default Flashcard
