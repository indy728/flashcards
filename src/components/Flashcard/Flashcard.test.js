import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Flashcard from './Flashcard'

configure({adapter: new Adapter()})

describe('Flashcard', () => {
    const flashcard = shallow(<Flashcard />)

    it('renders correctly', () => {
        expect(flashcard).toMatchSnapshot()
    })

    it('initializes `state` with reveal set to false', () => {
        expect(flashcard.state().reveal).toEqual(false)
    })
})