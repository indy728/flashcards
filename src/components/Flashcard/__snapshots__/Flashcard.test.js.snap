// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcard renders correctly 1`] = `ShallowWrapper {}`;
